### DISCUSSION TITLE
GHE Webhooks aren't working

### DISCUSSION THREAD

**Discussion by ELatestark** [WeblateOrg/weblate#12399] [Q&A] [answered]
Hey, I'm trying to set up Weblate using docker compose.
Now I'm stuck with GHE webhooks: they arent working.
My webhook setup:
![Снимок экрана от 2024-09-03 13-31-05](https://github.com/user-attachments/assets/c9600489-ab36-45ab-99c9-bf3637ae25af)

GHE says that my request is timed out, which basically means that [my server is not responding](https://docs.github.com/en/webhooks/testing-and-troubleshooting-webhooks/troubleshooting-webhooks#timed-out).
![Снимок экрана от 2024-09-02 22-08-35](https://github.com/user-attachments/assets/b7215108-56dc-44e7-a6cc-69bbfaea24a1)
I can see incoming http trafic in the moment I push to my directory and it triggers webhook, which means I set up Weblate in a wrong way.
![Снимок экрана от 2024-09-02 22-09-43](https://github.com/user-attachments/assets/3b413540-ab6b-497c-96a1-2f1f826674c3)
Things I've tried so far:

- set WEBLATE_ENABLE_HOOKS: 1, which, [according to documentation](https://docs.weblate.org/en/latest/admin/config.html#std-setting-ENABLE_HOOKS) enables anonymous hooks and should be enough to make it work.
- set WEBLATE_REQUIRE_LOGIN: 1 just to set WEBLATE_LOGIN_REQUIRED_URLS_EXCEPTIONS: /hooks/(.*)$

Could somebody tell me what am I doing wrong?

**Comment by nijel** [comment, marked as answer]
Is your Weblate running on `http`? Isn't it `https`? Timeout really looks like the request is not reaching Weblate at all (check docker logs, these include nginx requests logging).

**Reply by ELatestark** [reply]
> Is your Weblate running on `http`? Isn't it `https`? Timeout really looks like the request is not reaching Weblate at all (check docker logs, these include nginx requests logging).

Is it required to run Weblate on https to trigger webhooks? It's not a typo, I'm running Weblate with no certs, http only.
When I run `curl -X POST http://serverIP/hooks/github/` I get nginx logs that my curl is recieved.
![Снимок экрана от 2024-09-03 17-29-32](https://github.com/user-attachments/assets/5f69f1de-9844-4fab-8e9f-1512861f6ba5)

When I trigger webhook by pushing something to my repo I got nothing on nginx in docker, but still I can see related http traffic on port 80 of my host machine.

**Reply by nijel** [reply]
https is not required, I was just verifying it is correct. As the request from GitHub doesn't appear in the log, it might be some networking issue. Using `/etc/hosts`? Is Weblate listening on the interface which is accessed by GitHub?

**Reply by ELatestark** [reply]
So, turns out it was related to a network issue and VPN routes I had on my instance.
My question is irrelevant. Thanks for help tho.
